core/src/main/java/com/homeaway/streamingplatform/StreamRegistryApplication.java

```diff
@@ -168,7 +166,10 @@ public void run(final StreamRegistryConfiguration configuration, final Environme
         StreamValidator streamValidator = loadValidator(configuration, httpClient, regionDao);
         Preconditions.checkState(streamValidator != null, "streamValidator cannot be null.");
 
-        StreamDao streamDao = new StreamDaoImpl(managedProducer, managedKStreams, env, regionDao, infraManager, kafkaManager, streamValidator);
+        SchemaRegistrar schemaRegistrar = loadSchemaRegistrar(configuration);
+        Preconditions.checkState(schemaRegistrar != null, "schemaRegistrar cannot be null.");
```

Comment:
why not perform this check/initialization during initialization ?

Reply:
Also... if you add this as a post-condition check in `loadSchemaRegistrar` you don't need to do the check here.

* See: https://github.com/homeaway/stream-registry/pull/41/files#diff-a74858465e2f75329645cef3bd8c5ca3R230
* See: https://github.com/homeaway/stream-registry/pull/41/files#diff-a74858465e2f75329645cef3bd8c5ca3R232